ner_v1/detectors/textual/name/name_detection.py

```diff
@@ -380,30 +382,30 @@ def context_check_botmessage(self, botmessage):
         botmessage = regex_pattern.sub(r'', botmessage)
 
         botmessage = " " + botmessage.lower().strip() + " "
-        for variant in NAME_VARIATIONS.get(self.language, []):
-            if " " + variant + " " in botmessage:
+        for variant in PREVIOUS_MESSAGE_VARIATIONS.get(self.language, []):
+            if variant.lower() in botmessage:
```

Comment:
1. why remove the spaces? the spaces were added to ensure that checks follow word boundaries. e.g. "name" should not match "nickname"
2. We can lower the variants beforehand in the constants file to optimize

Reply:
1. Why don't we want to match with nickname/ first name/ surname, etc cases?
2.  Multiple languages where name is just a part of the token (eg. in German root word 'heiss' can be used as 'heissen/'heisst' etc depending on the kind of noun we are using with it.)
3. In case of French- we usually don't have space boundary around the 'nom' token as it comes towards the end of the sentence.
4. Such differences can occur throughout languages so I feel that we shouldn't have strict rules for it.

Let me know if I should still think up a different approach to satisfy all of these conditions @chiragjn